search/src/RawImage.cpp

```diff
@@ -432,16 +431,28 @@ RawImage createMedianImage(const std::vector<RawImage>& images)
     {
         for (int y = 0; y < height; ++y)
         {
+            int num_unmasked = 0;
             for (int i = 0; i < num_images; ++i)
             {
+                // Only used the unmasked pixels.
                 float pixVal = images[i].getPixel(x, y);
-                if ((pixVal == NO_DATA) || (isnan(pixVal))) pixVal = 0.0;
-                pixArray[i] = pixVal;
+                if ((pixVal != NO_DATA) && (!isnan(pixVal)))
+                {
+                    pixArray[num_unmasked] = pixVal;
+                    num_unmasked += 1;
+                }
+            }
+
+            if (num_unmasked > 0)
+            {
+                int median_ind = num_unmasked / 2;
+                std::nth_element(pixArray.begin(),
+                                 pixArray.begin() + median_ind,
+                                 pixArray.begin() + num_unmasked);
+                result.setPixel(x, y, pixArray[median_ind]);
+            } else {
+                result.setPixel(x, y, NO_DATA);
```

Comment:
How does leaving pixels with `NO_DATA` in the stamps affect filtering? That's the only change here I see that affects things further down the pipeline.

Reply:
That's a good point. The maximum value based filtering will work fine, but the CNN filtering might not work correctly anymore. I changed this to return a result of 0.0 in the case of all the pixels being no data to stay consistent with the old code.